### PR TITLE
fix anchor in token.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @vicancy @hellosnow

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 1]
```diff
@@ -342,7 +342,7 @@ private void UpdateHref(HtmlAgilityPack.HtmlNode link, string attribute, IDocume
             var originalHref = link.GetAttributeValue(attribute, null);
             var anchor = link.GetAttributeValue("anchor", null);
             link.Attributes.Remove("anchor");
-            var path = RelativePath.TryParse(originalHref);
+            var path = RelativePath.TryParse(UriUtility.GetPath(originalHref));
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >UriUtility.GetPath(originalHref) [](start = 45, length = 32)

to a local var

**Comment by vicancy** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 2]
```diff
@@ -221,15 +221,21 @@ private MarkupResult ParseCore(MarkupResult markupResult, FileAndType ft)
...
+                        string anchorInHref;
                         if (!string.IsNullOrEmpty(anchor) &&
                             string.Equals(link.Name, "href", StringComparison.OrdinalIgnoreCase))
                         {
-                            pair.Node.SetAttributeValue("anchor", anchor);
+                            anchorInHref = anchor;
                         }
+                        else
+                        {
+                            anchorInHref = null;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >anchorInHref [](start = 28, length = 12)

set null as initial value and can remove 'else'?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**